### PR TITLE
G7 Classic Status page cleanup

### PR DESCRIPTION
**Pull request by Navid200**
There are three items on the classic status page that are not useful for G7.
The reason is that the G7 Bluetooth name is not related to the transmitter ID as it was with G6.

How many people are tapping on forget device thinking that they are unpairing the transmitter?
I did!

If some day, we can figure out a workaround, we can bring them back.
But, for now, please approve removing them.

Screenshot_20231218-174712

### CODE REVIEW

**Comment by jamorham** [comment]
So in principle this is a good change I think but I would like you to use some better helper methods to declutter things and avoid duplication.

I will make a change for `shortTxId()` to be public static, so just use that instead of repeating the check in your own code.

Additionally I will include support for `showIfTrue` in to that layout, so you can do: app:showIfTrue="@{ms.using_g7()}" to avoid manually handling the visibility.

**Comment by Navid200** [comment]
Will be tested before tomorrow.

**Comment by Navid200** [comment]
Tested.  Looks OK.
The static page looks as shown above.

**Comment by Navid200** [comment]
I have made the change you asked for and I have tested it.
This is ready for merge.

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java [hunk 1]
```diff
@@ -28,12 +31,20 @@ public String gs(String id) {
...
     public boolean xmitterBattery() {
         return DexCollectionType.usesClassicTransmitterBattery();
     }
+
+    @Override
+    public boolean using_g7() { // True if we are using G7
+        if (DexCollectionType.getDexCollectionType() == DexCollectionType.DexcomG5 && Pref.getBooleanDefaultFalse("using_g6") && shortTxId()) { // If using G7
+            return true;
+        }
+        return false;
```
Comment (DiBochev) [inline app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java:48]: return DexCollectionType.getDexCollectionType() == DexCollectionType.DexcomG5 && Pref.getBooleanDefaultFalse("using_g6") && shortTxId()

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatus.java [hunk 2]
```diff
@@ -12,4 +12,6 @@ public interface MicroStatus {
 
     boolean xmitterBattery();
 
+    boolean using_g7(); // True when using G7
```
Comment (DiBochev) [inline app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatus.java:15]: not java style method naming. Java style method naming is camel case
also use java doc for "True when using G7"

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java [hunk 3]
```diff
@@ -28,12 +31,20 @@ public String gs(String id) {
     }
 
     @Override
-    public boolean bluetooth() {
-        return DexCollectionType.hasBluetooth();
+    public boolean bluetooth() { // Dexcom with Bluetooth except G7
+        return DexCollectionType.hasBluetooth() && !using_g7();
```
Comment (DiBochev) [inline app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java:35]: not sure if you are braking logic for other sensors with the extra check

Comment (Navid200) [reply app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java:35]: I don't know what you mean by that.  Please explain.

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java (hunk 1 above)
Comment (Navid200) [reply app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java:48]: Thanks a lot.  Fixing it in a minute.

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java (hunk 3 above)
Comment (Navid200) [reply app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java]: I see what you mean now.  
First, let me thank you for spending the time to review.

Please let me explain this concern.
There are only two possibilities under all circumstances.  Either we are using G7 or not.

If we are not using G7, what is being anded is true.  Anding true with anything changes nothing.  Therefore, nothing changes with respect to other sensors.

If we are using G7, the returned value becomes false, which removes three lines from the status page.

**Comment by Navid200** [comment]
Tested again after the syntax correction.

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java (hunk 3 above)
Comment (Navid200) [reply app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java]: Wow!  You have a point.  There is a third case of the value being unknown.  I need to make sure if the value is unknown, it doesn't break anything.  Thanks so much.

Comment (Navid200) [reply app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java]: @DiBochev Is it OK now?

**Comment by Navid200** [comment]
I'm now using ShortTxId.  Is anything else needed?

File: app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java [hunk 4]
```diff
@@ -28,12 +31,21 @@ public String gs(String id) {
...
     public boolean xmitterBattery() {
         return DexCollectionType.usesClassicTransmitterBattery();
     }
+
+    @Override
+    public boolean usingG7() {
+        try { // True if we are using G7
+            return DexCollectionType.getDexCollectionType() == DexCollectionType.DexcomG5 && Pref.getBooleanDefaultFalse("using_g6") && shortTxId();
+        } catch (Exception e) { // If there are any unknowns, show that we are not using G7
+            return false;
```
Comment (DiBochev) [inline app/src/main/java/com/eveningoutpost/dexdrip/ui/MicroStatusImpl.java:48]: just a suggestion maybe it is good idea to add debug log here

**Comment by Navid200** [comment]
We now have a best collector option of of G7.
So, to avoid code duplication, I am using that.

I have tested this after this last change with both G7 and G6.  There is no change to the classic status page of G6.
For G7, please see the images below.

| Before | After |  
| -------- | ---- |  
| Screenshot_20240221-095004 | Screenshot_20240221-095115 |

**Comment by Navid200** [comment]
The option to forget device does not work for G7.  And that is why it is important to remove it because everyone is not aware that it does not work:

**Comment by Navid200** [comment]
@jamorham 
Please merge this also so that people don't think that forget device works for G7.

**Review by jamorham — APPROVED** [review]

**Pull request merged**